### DISCUSSION TITLE
Use icons to represent class names and make code more scannable

### DISCUSSION THREAD

**Discussion by gustavopch** [tailwindlabs/tailwindcss#7202] [Ideas]
People say Tailwind CSS clutters the code when classes are inlined. It's true, but still, most of us consider it a good deal due to collocation, etc. I have an idea of how to eliminate that problem though.

Imagine an IDE extension that would allow the following:
- You type `class="uppercase text-green-500"`.
- When you move the cursor to another line, the editor replaces the class names with icons that represent them: `class="🔠 🟢"` (using emoji just to make the idea clear; in reality, proper icons would be created for each thing).
- If you move the cursor back to that line, the icons are converted back to class names so you can edit them.

To make it clear, the file would never be touched. It'd be visual sugar only. Somewhat like what happens here, except the original class name would be hidden:
![image](https://user-images.githubusercontent.com/10941958/151049159-933e334d-53df-438e-b9de-aa81881ab08b.png)

I think it would be a huuuge thing because it's a lot easier to scan icons than text — that's why we use icons in traffic signs. Visual cluttering would be gone.

**Comment by olivermrose** [comment]
There's too much ambiguity that would come with icons. For example, `🟢` could mean `bg-green-500`, `text-green-500`, `border-green-500`, etc., and then how would you differentiate shades? And how do you represent something like `shadow-*`?

**Reply by gustavopch** [reply]
Ambiguity would be avoided by carefully designing distinct icons for each thing. Of course, I only used emoji to make it easier to understand. So, for example:
- `bg-green-500`: green square (filled)
- `text-green-500`: green "T" character
- `border-green-500`: green square (outline)

If you wanted to differentiate shades, you'd just move the cursor to the line and read the class name, run some command/shortcut in the editor to deactivate the whole thing, or maybe even simply hover the icon to see the details in a tooltip.

Inspiration for the icons could be taken from existing software like Microsoft Word, Google Docs, Figma, Illustrator, which already heavily depend on iconography for their UIs.
